### PR TITLE
[ML] increase the default value of xpack.ml.max_open_jobs from 20 to 512 for autoscaling improvements

### PR DESCRIPTION
**Pull request by benwtrent**
This commit increases the `xpack.ml.max_open_jobs` from 20 to 512. Additionally, it ignores nodes that cannot provide an accurate view into their native memory.

If a node does not have a view into its native memory, we ignore it for assignment.

This effectively fixes a bug with autoscaling. Autoscaling relies on jobs with adequate memory to assign jobs to nodes. If that is hampered by the `xpack.ml.max_open_jobs` scaling decisions are hampered.

### CODE REVIEW

File: x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/JobNodeSelector.java [hunk 1]
```diff
@@ -174,7 +169,7 @@ public JobNodeSelector(ClusterState clusterState,
                 continue;
             }
             // Assuming the node is eligible at all, check loading
-            allocateByMemory = currentLoad.isUseMemory();
+            boolean allocateByMemory = currentLoad.isUseMemory();
```
Comment (droberts195) [inline x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/JobNodeSelector.java:172]:
```suggestion
            boolean canAllocateByMemory = currentLoad.isUseMemory();
```

I think this variable should be renamed now that it's being used for a slightly different purpose.

**Review by droberts195 — APPROVED** [review]
LGTM

**Comment by lcawl** [comment]
This PR will affect https://www.elastic.co/guide/en/elasticsearch/reference/master/ml-settings.html. If you want me to add a commit to this PR for that change, just let me know!

**Comment by benwtrent** [comment]
@lcawl I think I updated the necessary places :D

**Pull request merged**